src/Diagnostics.RuntimeHost/Services/DiagnosticTranslator/DiagnosticTranslatorService.cs

```diff
@@ -0,0 +1,288 @@
+﻿using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Diagnostics.RuntimeHost.Services.CacheService;
+using System.Net.Http.Headers;
+using Diagnostics.ModelsAndUtils.Models;
+using System.Data;
+using Microsoft.CodeAnalysis;
+using Diagnostics.ModelsAndUtils.Attributes;
+
+namespace Diagnostics.RuntimeHost.Services.DiagnosticsTranslator
+{
+    public interface IDiagnosticTranslatorService
+    {
+        Task<Response> GetResponseTranslations(Response response, string language);
+        Task<IEnumerable<DiagnosticApiResponse>> GetMetadataTranslations(IEnumerable<DiagnosticApiResponse> listDetectorsResponse, string language);
+    }
+
+    public class DiagnosticTranslatorService : IDiagnosticTranslatorService
```

Comment:
Couple things:
1. We should have a "enabled" switch so that this can be managed depending on situation and environment.
2. Since we are caching string translations received from the translation API - which is basically caching content which double the size of the detector response, we should find a way to limit memory usage of this cache i.e. implement it as a LRU cache maybe.

Reply:
Could probably do a memory test of 20-30 detector calls on localhost and see the impact.